src/Browser/WebStorage.purs

```diff
@@ -12,30 +14,35 @@ module Browser.WebStorage
   , setItem
   ) where
 
+  import Control.Monad.Eff
   import Data.Maybe
+  import Data.Function
+
+  foreign import data WebStorage :: !
+  type EffWebStorage eff = Eff (webStorage :: WebStorage | eff)
 
   class Storage s where
-    clear :: s -> s
-    getItem :: forall v. s -> String -> Maybe v
-    key :: s -> Number -> Maybe String
-    length :: s -> Number
-    removeItem :: s -> String -> s
-    setItem :: forall v. s -> String -> v -> s
+    clear :: forall eff. s -> EffWebStorage eff Unit
+    getItem :: forall eff. s -> String -> EffWebStorage eff (Maybe String)
+    key :: forall eff. s -> Number -> EffWebStorage eff (Maybe String)
+    length :: forall eff. s -> EffWebStorage eff Number
+    removeItem :: forall eff. s -> String -> EffWebStorage eff Unit
+    setItem :: forall eff. s -> String -> String -> EffWebStorage eff Unit
```

Comment:
Well... I'm not following meaning of its return value, so I deleted it.
actually, Eff can execute side effect without return value. e.g. [`Debug.Trace.trace`](https://github.com/purescript/purescript/blob/master/prelude/prelude.purs#L990-L998)(and I forgot `return {}`. I'll add it soon.).

My rewritten api can use:

```
module Main where

import Control.Monad.Eff
import Browser.WebStorage
import Debug.Trace
import Data.Maybe

main = do
  length localStorage >>= print
  setItem localStorage "foo" "foo"
  setItem localStorage "bar" "bar"
  getItem localStorage "foo" >>= print
  key localStorage 0 >>= print
  length localStorage >>= print
  removeItem localStorage "foo"
  getItem localStorage "foo" >>= print
  clear localStorage
  length localStorage >>= print
```

I think it works well.
But if I don't understand value of return value, please tell me. I'll do rollback it.